### ISSUE TITLE
Unable to build docker image on specific branch/tag

### ISSUE THREAD

**Issue by altulek** [grafana/grafana#34780]
**What happened**:
Docker image can't be build on specific GIT tag

**What you expected to happen**:
I woluld like to build docker image on specific GIT tag

**How to reproduce it (as minimally and precisely as possible)**:
```
git.exe checkout v7.5.7
docker build -f .\Dockerfile .
...
[js-builder  5/11] RUN yarn install --pure-lockfile --no-progress:
#29 0.927 yarn install v1.22.5
#29 1.286 [1/5] Validating package.json...
#29 1.383 [2/5] Resolving packages...
#29 5.153 [3/5] Fetching packages...
#29 5.372 error Couldn't find the binary git
#29 5.372 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
------
executor failed running [/bin/sh -c yarn install --pure-lockfile --no-progress]: exit code: 1`
```

**Anything else we need to know?**:
The same happend with Dockerfile.ubuntu
It should be fixed in #33223, but it seems that it work only on main branch
**Environment**:
- Grafana version: 7.5.7
- OS Grafana is installed on: Docker
- User OS & Browser: Windows
- Docker: Docker version 20.10.5, build 55c4c88

**Comment by Elfo404** [comment]
looping in @grafana/backend-platform

**Comment by dsotirakis** [comment]
Hello @altulek, thanks for raising the issue, this change wasn't included into `v7.x.x` versions, that's why you cannot see it when checking out `v7.5.7` version (you'll see `RUN apk --no-cache add git` missing from `Dockerfile`) since for some reason image builds became a bit flakey when including git binary. Will be there for `v8` though!

**Comment by altulek** [comment]
Is any other way to build docker image with specific version? 
Only way I can do this is checkout to specific commit on main branch and then building. Is it correct way? 

Adding `RUN apk --no-cache add git` to Dockerfile are not resloving this issue, because probaly there is many other files witch are not included in 7.x.x

**Comment by dsotirakis** [comment]
By specific commit do you mean the commit on main branch when we released `v7.5.7`? This _should_ be fine.

What's the error you get if you add `RUN apk --no-cache add git`?

**Comment by altulek** [comment]
> By specific commit do you mean the commit on main branch when we released `v7.5.7`? This _should_ be fine.

Yes. Ok, I will try

> What's the error you get if you add `RUN apk --no-cache add git`?

It ended there:
```
docker build -f .\Dockerfile .
[+] Building 179.3s (19/36)
 => [internal] load build definition from Dockerfile                                                                                                                                                                                    0.1s
 => => transferring dockerfile: 2.55kB                                                                                                                                                                                                  0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                       0.0s
 => => transferring context: 35B                                                                                                                                                                                                        0.0s
 => [internal] load metadata for docker.io/library/alpine:3.13                                                                                                                                                                          0.7s
 => [internal] load metadata for docker.io/library/node:14.15.5-alpine3.13                                                                                                                                                              0.8s
 => [internal] load metadata for docker.io/library/golang:1.16.1-alpine3.13                                                                                                                                                             0.9s
 => [js-builder  1/12] FROM docker.io/library/node:14.15.5-alpine3.13@sha256:03b86ea1f9071a99ee3de468659c9af95ca0bedbcd7d32bf31d61fa32c1a8ab3                                                                                           0.0s
 => [internal] load build context                                                                                                                                                                                                       1.9s
 => => transferring context: 413.72kB                                                                                                                                                                                                   1.8s
 => [stage-2  1/10] FROM docker.io/library/alpine:3.13@sha256:69e70a79f2d41ab5d637de98c1e0b055206ba40a8145e7bddb55ccc04e13cf8f                                                                                                          0.0s
 => [go-builder 1/8] FROM docker.io/library/golang:1.16.1-alpine3.13@sha256:39a5f1284ccbb22cc2695a73d196d5d833383c1f073f520ab6c9360da84fc782                                                                                            0.0s
 => CACHED [go-builder 2/8] RUN apk add --no-cache gcc g++                                                                                                                                                                              0.0s
 => CACHED [go-builder 3/8] WORKDIR /go/src/github.com/grafana/grafana                                                                                                                                                                  0.0s
 => CACHED [go-builder 4/8] COPY go.mod go.sum ./                                                                                                                                                                                       0.0s
 => CACHED [go-builder 5/8] RUN go mod verify                                                                                                                                                                                           0.0s
 => CACHED [go-builder 6/8] COPY pkg pkg                                                                                                                                                                                                0.0s
 => CACHED [go-builder 7/8] COPY build.go package.json ./                                                                                                                                                                               0.0s
 => CACHED [js-builder  2/12] WORKDIR /usr/src/app/                                                                                                                                                                                     0.0s
 => CACHED [js-builder  3/12] COPY package.json yarn.lock ./                                                                                                                                                                            0.0s
 => CACHED [js-builder  4/12] COPY packages packages                                                                                                                                                                                    0.0s
 => [js-builder  5/12] RUN apk --no-cache add git                                                                                                                                                                                       3.1s
 => [go-builder 8/8] RUN go run build.go build                                                                                                                                                                                        176.2s
 => => # go: downloading github.com/googleapis/gax-go v2.0.2+incompatible
 => => # go: downloading github.com/googleapis/gax-go/v2 v2.0.5
 => => # go: downloading github.com/oklog/ulid v1.3.1
 => => # go: downloading go.uber.org/goleak v1.1.10
 => => # go: downloading go.opencensus.io v0.22.6
 => => # go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
 => [js-builder  6/12] RUN yarn install --pure-lockfile --no-progress                                                                                                                                                                 173.2s
 => => # info fsevents@1.2.9: The platform "linux" is incompatible with this module.
 => => # info "fsevents@1.2.9" is an optional dependency and failed compatibility check. Excluding it from installation.
 => => # [3/4] Linking dependencies...
 => => # warning " > eslint-plugin-import@2.20.2" has incorrect peer dependency "eslint@2.x - 6.x".
 => => # warning " > styled-components@5.1.1" has unmet peer dependency "react-is@>= 16.8.0".
 => => # [4/4] Building fresh packages...
```

**Comment by Andrewpk** [comment]
> Hello @altulek, thanks for raising the issue, this change wasn't included into `v7.x.x` versions, that's why you cannot see it when checking out `v7.5.7` version (you'll see `RUN apk --no-cache add git` missing from `Dockerfile`) since for some reason image builds became a bit flakey when including git binary. Will be there for `v8` though!

So were you using a different Dockerfile than what was being distributed to build the code internally? I imagine without this code change you would've also seen CI build errors?
